src/test/ui/rfc-2294-if-let-guard/feature-gate.rs

```diff
@@ -58,8 +58,10 @@ fn _macros() {
     }
     use_expr!((let 0 = 1 && 0 == 0));
     //~^ ERROR `let` expressions in this position are unstable
```

Comment:
I'm not a fan of the duplicated errors -- is it possible to suppress the later errors somehow?

Reply:
Maybe a follow-up.

Reply:
FWICT, `parse_let_expr` needs to short-circuit instead of emitting errors